test/functional/Java22andUp/src/org/openj9/test/jep454/downcall/StructTests2.java

```diff
@@ -2953,4 +2953,220 @@ public void test_add3DoubleStructs_returnStruct_2() throws Throwable {
 			Assert.assertEquals((double)doubleHandle3.get(resultSegmt, 0L), 88.579D, 0.001D);
 		}
 	}
+
+	@Test
+	public void test_addBoolsFromMultipleStructPtrs_returnStruct_2() throws Throwable {
+		GroupLayout structLayout = MemoryLayout.structLayout(JAVA_BOOLEAN.withName("elem1"), JAVA_BOOLEAN.withName("elem2"));
+		VarHandle boolHandle1 = structLayout.varHandle(PathElement.groupElement("elem1"));
+		VarHandle boolHandle2 = structLayout.varHandle(PathElement.groupElement("elem2"));
+
+		FunctionDescriptor fd = FunctionDescriptor.of(structLayout, ADDRESS, ADDRESS, ADDRESS);
+		MemorySegment functionSymbol = nativeLibLookup.find("addBoolsFromMultipleStructPtrs_returnStruct").get();
+		MethodHandle mh = linker.downcallHandle(fd, Linker.Option.critical(true));
+
+		try (Arena arena = Arena.ofConfined()) {
+			MemorySegment structSegmt1 = MemorySegment.ofArray(new byte[2]);
```

Comment:
We should have some tests that for the case where the offset from the base of an object is non-zero.

Reply:
I will add tests later to verify the situation.

Reply:
I am still working on the new tests with non-zero offset.

Reply:
I added new tests with `asSlice()` with the non-zero offset specified which work good with the latest code changes.